src/Common/src/TypeSystem/Common/SignatureVariable.cs

```diff
@@ -64,7 +64,7 @@ public override int GetHashCode()
 
         protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
         {
-            TypeFlags flags = 0;
+            TypeFlags flags = TypeFlags.ContainsGenericVariablesComputed | TypeFlags.ContainsGenericVariables;
 
```

Comment:
Calling .ContainsGenericVariables on TypeDescs for things like List<!0> was failing with an assert because of the SignatureVariable's .ContainsGenericVariables.